components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesUseCases.kt

```diff
@@ -20,6 +29,10 @@ class TopSitesUseCases(topSitesStorage: TopSitesStorage) {
          */
         operator fun invoke(title: String, url: String, isDefault: Boolean = false) {
             storage.addTopSite(title, url, isDefault)
+
+            scope.launch {
+                emitCountFact(storage.getTopSitesCount())
+            }
```

Comment:
How come we need a Coroutine? Isn't `emitCountFact ` synchronous?

Reply:
the `storage.getTopSitesCount()` is a database call so it should not be a blocking call.

Reply:
gotcha